Sources/DVR/Session.swift

```diff
@@ -134,7 +139,7 @@ open class Session: URLSession {
     }
 
     func finishTask(_ task: URLSessionTask, interaction: Interaction, playback: Bool) {
-        needsPersistence = needsPersistence || !playback
+        needsPersistence = (needsPersistence || !playback) && recordMode != .all
```

Comment:
So if `recordMode == .all` then `needsPersistence` would be false, but if `.all` is effectively our re-record, then I don't see how this works.

Reply:
.all isn't re-record. .all is live. .once is record (and re-record). It's based off when I originally asked you about VCR having 4 playback modes while our TEST_MODE environment variable only has 3. 
.all = live
.once = record
.none = playback
.newEpisode isn't used since we only have 3 TEST_MODE options

Reply:
I think we should re-think our TEST_MODE options